design/nfs.md

```diff
@@ -0,0 +1,157 @@
+
+### Initial Setup
+
+The flow of creating NFS backed storage in Rook is
+1. The settings are determined and saved in an NFS server CRD (rook-nfs.yaml)
+2. `kubectl create -f rook-nfs.yaml`
+3. Rook is notified of the new CRD
+   - Rook starts the NFS daemon on the desired nodes in the cluster
+   - Rook creates a storage class
+4. The pod which requires NFS storage creates a PVC to consume the volume.
+
+### Deployed service
+
+When the NFS CRD instance is created, Rook responds to this request by starting the NFS daemon with the configuration and exports stated in the CRD and creates a service to expose NFS.
+
+### NFS CRD
+
+The NFS CRD spec will specify the following:
+1. NFS server configuration
+    - export (The directory being exported)
+    - host (The host or network to which the export is being shared)
+    - options (The options to be used for the host) e.g., read and write permission, sync, root_squash.
+2. NFS server storage backend configuration. E.g., configuration for various storage backends(ceph, ebs, azure disk etc) that will be shared using NFS.
+
+The parameters to configure NFS are:
+
+| Parameter                       | Description                                                              | Default              |
+|---------------------------------|--------------------------------------------------------------------------|----------------------|
+| `replicas`                      | The no. of NFS daemon to start                                            | `1`                  |
+| `serverConfig`                  | NFS server configuration parameters                                      | <none>               |
+| `serverConfig.validHosts`       | The host or network to which export is being shared.                     | Current pod network  |
+| `serverConfig.accessMode`       | Reading and Writing permissions on the share.                             | `RO`                 |
+| `serverConfig.rootSquash`       | This prevents root users connected remotely from having root privileges. | `true`               |
+| `serverConfig.sync`             | The NFS server will not reply to requests before changes made by previous requests are written to disk. | `true` |
+| `volumes`                       | Attach a volume to NFS server Pod for sharing                            | <none>               |
+| `volumes.name`                  | Name of the volume that will be attached to NFS server pod.  | <none>               |
+| `volumes.persistentVolumeClaim` | Claim to get volume(Volume could come from hostPath, cephFS, cephRBD, googlePD, EBS etc.). | <none> |
+| `volumes.claimName`             | Name of the PVC.                                                         | <none> |
+
+### Examples:
+
+1. For sharing a volume(could be hostPath, cephFS, cephRBD, googlePD, EBS etc.) using NFS, which will be shared as /myshare by the NFS server whose Nfs-Ganesha export entry looks like:
```

Comment:
This example just uses EBS. I'd suggest you mention all the other types (hostPath, cephFS, cephRBD, etc) as part of the table above, perhaps under the `claimName` property.

Reply:
`claimName` is to provide name for PCV from where to attach volume to NFS for sharing, so it can be any name, will it be good to mention type of volume being attached to NFS server pod eg., `volumes.persistentVolumeClaim.volumeType` ? But i don't think this option will be useful.